react/src/components/FormItemWithUnlimited.tsx

```diff
@@ -0,0 +1,66 @@
+import Flex from './Flex';
+import { Form, FormInstance } from 'antd';
+import Checkbox, { CheckboxChangeEvent } from 'antd/es/checkbox';
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+
+interface FormItemWithUnlimitedProps {
+  form: FormInstance;
+  fieldName: string | Array<string>;
+  checkboxFieldName: string;
+  label?: string;
+  fallbackValue?: any;
+  render: (disabled: boolean) => React.ReactNode;
+}
+
```

Comment:
Great! But we can improve some parts:
- Remove `checkboxFieldName` and add `unlimitedValue`.
- Use `name` instead of `fieldName`.
- Minor: Use `children` with `cloneElement` instead of a `render` function to match the Form.Item patterns.
- Use `form.resetFields(['someName'])` instead of `fallbackValue`.
- Please create `FormItemWithUnlimitedProps` by extending `FormItemProps`.